crates/shared/src/test_utils/output_assert.rs

```diff
@@ -51,27 +51,29 @@ fn assert_output_contains(output: &str, lines: &str) {
     let asserted_lines: Vec<String> = lines.lines().map(std::convert::Into::into).collect();
     let mut actual_lines: Vec<String> = output.lines().map(std::convert::Into::into).collect();
 
-    let mut matches = true;
+    let mut contains = true;
     let mut out = String::new();
 
     for line in &asserted_lines {
         if is_present(line, &mut actual_lines) {
             out.push_str("| ");
         } else {
-            matches = false;
+            contains = false;
             out.push_str("- ");
         }
         out.push_str(line);
         out.push('\n');
     }
-    for remaining_line in actual_lines {
-        matches = false;
-        out.push_str("+ ");
-        out.push_str(&remaining_line);
-        out.push('\n');
+
+    if !contains {
+        for remaining_line in actual_lines {
+            out.push_str("+ ");
+            out.push_str(&remaining_line);
+            out.push('\n');
+        }
     }
```

Comment:
nit: could probably use an iter().foreach() with a closure instead

```suggestion
    if !contains {
      actual_lines.iter().for_each(|line| out.push_str(&format!("+ {}\n", line)));
    }
```

Reply:
Updated this, but I'm not sure it's more readable this way 😅